libraries/ESP8266WiFi/src/WiFiClientSecureBearSSL.cpp

```diff
@@ -179,7 +179,10 @@ void WiFiClientSecure::stop() {
     _client->abort();
   }
   WiFiClient::stop();
-  _clearAuthenticationSettings();
+  // Only if we've already connected, clear the connection options
+  if (_handshake_done) {
```

Comment:
What happens if I set up a connection, attempting to connect fails (_handshake_done never becomes true), and then decide to abort and call stop()? Won't the flags remain?
What if the certs are then changed and a different connection is attempted? won't the previous flags be still there?